### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by royassis** [hashicorp/terraform-provider-aws#29094]
### Terraform Core Version

1.3.7

### AWS Provider Version

v4.30.0

### Affected Resource(s)

I'm using Terraform cloud and using the **aws_api_gateway_integration** resources.
When applying the resources **connection_type** attribute always change from "" => **"INTERNET"**. 

Event if the attribute is already set to "INTERNET"

### Expected Behavior

No change need to be made

### Actual Behavior

Changes on apply

### Terraform Configuration Files

```terraform
module "lambda_function_GetApigwResources" {
  source = "terraform-aws-modules/lambda/aws"

  function_name            = "GetApigwResources"
  handler                  = "app.handler"
  runtime                  = "python3.9"
  timeout                  = 10
  store_on_s3              = true
  s3_bucket                =<redacted>
  s3_prefix                = "lambda-builds/"
  attach_policy_statements = true
  policy_statements        = {
    apigateway = {
      effect    = "Allow",
      actions   = ["apigateway:Get*"],
      resources = ["*"]
    }
  }
  role_path   = <redacted>
  policy_path =<redacted>
  source_path = <redacted>
  tags        = {
    <redacted>= <redacted>
  }
  publish          = true
  allowed_triggers = local.allowed_triggers
}

resource "aws_api_gateway_rest_api" "api" {
  api_key_source               = "HEADER"
  binary_media_types           = []
  disable_execute_api_endpoint = false
  minimum_compression_size     = -1
  name                         = "somename"
  policy                       = jsonencode(
    {
      Statement = [
        {
          Action = [
            "execute-api:Invoke",
            "cloudfront:updateDistribution",
          ]
          Effect    = "Allow"
          Principal = {
            AWS = "*"
          }
          Resource = "*"
        },
      ]
      Version = "2012-10-17"
    }
  )
  tags     = {}
  tags_all = {}

  endpoint_configuration {
    types = [
      "PRIVATE",
    ]
    vpc_endpoint_ids = [
      "<vpc_endpoint_id >",
    ]
  }
}

resource "aws_api_gateway_resource" "index" {
  rest_api_id = aws_api_gateway_rest_api.api.id
  parent_id   = aws_api_gateway_rest_api.api.root_resource_id
  path_part   = "index"
}


resource "aws_api_gateway_integration" "index" {
  resource_id = aws_api_gateway_resource.index.id
  rest_api_id = aws_api_gateway_rest_api.api.id
  http_method = aws_api_gateway_method.index.http_method

  timeout_milliseconds    = 29000
  cache_namespace         = aws_api_gateway_resource.index.id
  integration_http_method = "POST"
  type                    = "AWS_PROXY"
  connection_type         = "INTERNET"
  uri                     = module.lambda_function_GetApigwResources.lambda_function_invoke_arn
}
```

### Steps to Reproduce

run

### Would you like to implement a fix?

None

**Comment by mattburgess** [comment]
Duplicate of #28997.

**Comment by justinretzolk** [comment]
Hey @royassis 👋 Thanks for taking the time to raise this! It looks like this was resolved with #29016, with will go out with `4.52.0` in just a few minutes. Given that, we'll close this one out.
